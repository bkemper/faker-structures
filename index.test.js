const faker = require('faker');
const { collection, record } = require('.');

// see, https://github.com/Marak/faker.js#setting-a-randomness-seed
faker.seed(1234567890);

describe('fakerStructure', () => {
  describe('.collection', () => {
    it('with faker functions', () => {
      const subject = collection({
        id: faker.random.number,
        firstName: faker.name.firstName,
        lastName: faker.name.lastName
      });

      expect(subject(3)).toMatchSnapshot();
    })

    it('with custom functions', () => {
      const subject = collection({
        name: () => `${faker.name.firstName()} ${faker.name.lastName()}`
      });

      expect(subject(3)).toMatchSnapshot();
    })
  })

  describe('.record', () => {
    it('with faker functions', () => {
      const subject = record(faker.random.number, {
        id: faker.random.number,
        firstName: faker.name.firstName,
        lastName: faker.name.lastName
      });

      expect(subject(3)).toMatchSnapshot();
    })

    it('with custom functions', () => {
      const subject = record(faker.random.number, {
        name: () => `${faker.name.firstName()} ${faker.name.lastName()}`
      });

      expect(subject(3)).toMatchSnapshot();
    })
  })
})
