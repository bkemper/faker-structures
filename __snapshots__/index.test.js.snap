// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fakerStructure .collection with custom functions 1`] = `
Array [
  Object {
    "name": "Rosalyn Sanford",
  },
  Object {
    "name": "Cecelia Gleason",
  },
  Object {
    "name": "Chaya Zieme",
  },
]
`;

exports[`fakerStructure .collection with faker functions 1`] = `
Array [
  Object {
    "firstName": "Eva",
    "id": 61879,
    "lastName": "McLaughlin",
  },
  Object {
    "firstName": "Sibyl",
    "id": 14903,
    "lastName": "Feeney",
  },
  Object {
    "firstName": "Holden",
    "id": 89165,
    "lastName": "Klocko",
  },
]
`;

exports[`fakerStructure .record with custom functions 1`] = `
Object {
  "13602": Object {
    "name": "Lexi Boehm",
  },
  "2891": Object {
    "name": "Henri Heathcote",
  },
  "78563": Object {
    "name": "Alexandra Hansen",
  },
}
`;

exports[`fakerStructure .record with faker functions 1`] = `
Object {
  "20705": Object {
    "firstName": "Quentin",
    "id": 40099,
    "lastName": "Purdy",
  },
  "78638": Object {
    "firstName": "Eusebio",
    "id": 40518,
    "lastName": "Wiegand",
  },
  "932": Object {
    "firstName": "Aurelie",
    "id": 54009,
    "lastName": "Leuschke",
  },
}
`;
